llvm-bazel/llvm-project-overlay/mlir/tblgen.bzl

```diff
@@ -27,19 +29,33 @@ def _get_transitive_srcs(srcs, deps):
       a collection of the transitive sources
     """
     return depset(
-        srcs,
+        direct = srcs,
         transitive = [_get_dep_transitive_srcs(dep) for dep in deps],
     )
 
 def _get_transitive_includes(includes, deps):
     return depset(
-        includes,
+        direct = includes,
         transitive = [_get_dep_transitive_includes(dep) for dep in deps],
     )
 
+def _process_includes(ctx, raw_includes):
```

Comment:
I wonder if there's a more intention revealing name, wdyt about resolve_includes?

Reply:
Sure. I'm also trying to find a better way to get the current package path and current repository path. `native.package_name()` and `native.repository_name()` can only be called during loading. Basically I want to be able to replicate the includes parameters from `cc_library` but AFAICT that isn't actually implemented in Starlark. Like `strip_include_prefix` treats absolute paths as relative to the repository (https://docs.bazel.build/versions/master/be/c-cpp.html#cc_library.strip_include_prefix), but I can't figure out how to do that. Is there something that has this functionality implemented in Starlark?

Reply:
Ok similarly, the way I have this implemented doesn't work. I'm testing this out in IREE and the package_path I'm getting is just `llvm-project/mlir/` as opposed to `external/llvm-project/mlir/`. How do I get the actual path here? The documentation for root says that they are "are mapped together into a single directory tree to form the execution environment"
(https://docs.bazel.build/versions/master/skylark/lib/root.html), which suggests to me that I shouldn't have to be doing all this futzing with various paths, but that doesn't seem to be the case.

Reply:
I came up with the solution of adding the BUILD file as a private dep in the rule

```bzl
"_build_file": attr.label(default = ":BUILD", allow_single_file = True),
```

is that the best we've got?

Reply:
I found `ctx.label`. I still feel like I'm doing a silly amount of path munging here.